### ISSUE TITLE
[FEATURE REQUEST] JetBookings, check-in / check-out times with accurate timing

### ISSUE THREAD

**Issue by makovane** [Crocoblock/suggestions#1597]
Case for rental bookings. At the time the minimum booking time available is 1 day. I would like the rental bookings to be made with more accurate check-in and check-out times (15 min intervals). This is the basic functionality found e.g. on any car rental site (rentalcars.com).

**Comment by datatime-sk** [comment]
Please, this is very necessary and we need it too. +1

**Comment by TIBINTCO** [comment]
This is very important for us as well. Woocommerce bookings plugin does this very well. With time optimization for the business with:

- Max Bookings per block
- Min Block bookable
- Max Block Bookalbe
- Requried a buffer period
- Set daes availability 
- check rules against...
- first block starts at:
- Range type
- Range
- Bookable
- and, PRIORITY.

Here is an example: [WooCommerce Bookings Backend](https://paste.pics/ebb2cf5d6491fdeaee4d212e3e5c9b33)

**Comment by mrroy119** [comment]
Yes, please implement this feature as soon as possible because it's completely useless without time functionality....................

**Comment by tamas5** [comment]
It would be neccessery for us too. Please implement it!

**Comment by nickbeus** [comment]
Yes, Dito! For Car Rental sites it's very important item to have. Return time is essential and another field for the minimal period of time to clean the car etc., so the new booking can be delayed. But I have a feeling Crocoblock is almost done with this feature ;-)

**Comment by nickbeus** [comment]
We need this also. A pickup and return time + car cleaning time (buffer) = new rental pickup time (instead next day)

**Comment by dalexdesigngroup** [comment]
We agree this is critical feature for rental based businesses - We have customer selected check-in and check-out times for storage and need to be able to track / limit when a person can book based on store hours ( jet appointments does not fit our needs either, Jet booking is the closest)

**Comment by Qubadi** [comment]
1+

**Comment by ihslimn** [comment]
+1

the customer must have the choice to book:

- For one night
- For the afternoon
- For the weekend

I've managed to do it for one night or for the weekend but I can't do it for the afternoon, knowing that another customer can book the same day at night even if a customer has booked the afternoon.

**Comment by marowi85** [comment]
+1

I need this as well.

Let's assume you have a website for trailer rental. One customer only needs the trailer for a few hours, another customer wants to rent it for several days. In this case, it would be better to select a start and end date with time on the booking page.

**Comment by hichamdeveloper** [comment]
Is there any update, or do you consider it on your roadmap? Thank you.

**Comment by alvarindev** [comment]
any plans for this?

**Comment by TheBigBanik** [comment]
Absolutely a must have feature.

**Comment by alvarindev** [comment]
I've solved it by adding extra fields and extra columns in the database and then some JS to adjust calculations BUT I can't filter by time. For me works fine like this but I imagine that could be not 100% useful for others

**Comment by maliktanveerdhool** [comment]
> I've solved it by adding extra fields and extra columns in the database and then some JS to adjust calculations BUT I can't filter by time. For me works fine like this but I imagine that could be not 100% useful for others

Hi can you please share your code how to integrate extra columns?

**Comment by aalexandr21** [comment]
was added in JetBooking 3.7.0
https://youtu.be/7oMuv0vxTjw?si=7t44OgpASvM_R3NQ
